.circleci/config.yml

```diff
@@ -46,6 +52,12 @@ jobs:
             set -x
             source ~/project/docker/commands.bash
             cdash_run_and_submit_pgsql_ctest
+  build_docker:
```

Comment:
Putting this here because I can't comment on lines you didn't edit.

1. The `docker compose up ...` command builds and tags a `testing` image.  Since that image doesn't carry over to this job, you need to rebuild it before attempting to push the testing image.  You could possibly do an initial testing build and use the result of that in the other jobs.  I wasn't able to find a clear CircleCI example for this, but it seems possible to do.  See the information about "fan-out/fan-in" workflows and "using workspaces to share data between jobs" [here](https://circleci.com/docs/workflows/).

2. You could use CircleCI's automatic filtering to only run the publish step on merge to master/tag.  As it is now, I believe the `testing` and `latest` images will actually get overwritten when we create a new release, meaning that latest and testing are not actually the latest until the next PR is merged into master.